spec/features/user_edit_spec.rb

```diff
@@ -1,77 +1,95 @@
+          end
+        end
 
-      it 'redirects you to the staff dashboard' do
-        click_on 'Save changes'
-        expect(page.current_url).to eql staff_dashboard_url
-      end
+        it 'only changes the given attributes' do
+          click_on 'Save changes'
+          expect(user.reload)
+            .to have_attributes first_name: 'Bananas', last_name: 'Smith'
+        end
 
-      it 'gives a flash message' do
-        expect_flash_message(:user_update)
-        click_on 'Save changes'
-      end
-    end
+        it 'redirects you to the staff dashboard' do
+          click_on 'Save changes'
+          expect(page.current_url).to eql staff_dashboard_url
+        end
 
-    context 'a field is left blank' do
-      let!(:attributes) { attributes_for(:user).except :staff }
-      before :each do
-        within 'form.edit_user' do
-          fill_in_fields_for User, attributes: attributes
-            .merge(first_name: '')
+        it 'gives a flash message' do
+          expect_flash_message(:user_update)
+          click_on 'Save changes'
         end
       end
 
-      it 'does not change the blank attribute' do
-        click_on 'Save changes'
-        expect(user.reload.attributes.symbolize_keys)
-          .to include first_name: user.first_name, last_name: user.last_name
-      end
+      context 'a field is left blank' do
+        let!(:attributes) { attributes_for(:user).except :staff }
+        before :each do
+          within 'form.edit_user' do
+            fill_in_fields_for User, attributes: attributes
+              .merge(first_name: '')
+          end
+        end
 
-      it 'redirects back to the edit user page' do
-        click_on 'Save changes'
-        expect(page.current_url).to eql edit_user_url(user)
-      end
+        it 'does not change the blank attribute' do
+          click_on 'Save changes'
+          expect(user.reload.attributes.symbolize_keys)
+            .to include first_name: user.first_name, last_name: user.last_name
+        end
+
+        it 'redirects back to the edit user page' do
+          click_on 'Save changes'
+          expect(page.current_url).to eql edit_user_url(user)
+        end
 
-      it 'has flash errors' do
-        click_on 'Save changes'
-        expect(page).to have_selector '#errors',
-                                      text: "First name can't be blank"
+        it 'has flash errors' do
+          click_on 'Save changes'
+          expect(page).to have_selector '#errors',
+                                        text: "First name can't be blank"
+        end
       end
     end
   end
+  context 'with staff privilege' do
+    before :each do
+      when_current_user_is :staff, integration: true
+    end
+    let!(:user) { create :user, staff: true, last_name: 'Smith' }
+    it 'does not have link to page' do
+      visit staff_dashboard_url
+      expect(page).not_to have_link "#{user.last_name}, #{user.first_name}"
```

Comment:
There's [a method](https://github.com/umts/jobapps/commit/0086c0a8c551c8b14c84795dd8f645f9862f2433#diff-4676c008b11a5480d73d4a6de01e45b9R20) for this.